pkg/kubectl/cmd/create/create_clusterrole.go

```diff
@@ -174,3 +197,22 @@ func (c *CreateClusterRoleOptions) RunCreateRole() error {
 
 	return c.PrintObj(clusterRole)
 }
+
+func generateAggretaionRule(rules []string) (*rbacv1.AggregationRule, error) {
+	matchLabels := map[string]string{}
+	for _, rule := range rules {
+		parts := strings.Split(rule, "=")
```

Comment:
You can just use the utilities in staging/src/k8s.io/apiserver/pkg/util/flag/map_string_string.go